src/BloomExe/ImageProcessing/ImageServer.cs

```diff
@@ -75,20 +77,33 @@ protected override bool ProcessRequest(IRequestInfo info)
 				return false;
 
 			var r = GetLocalPathWithoutQuery(info);
-			if (r.EndsWith(".png") || r.EndsWith(".jpg"))
+
+			// only process these image types
+			if (!r.EndsWith(".png") && !r.EndsWith(".jpg") && !r.EndsWith(".svg")) return false;
+
+			r = r.Replace("thumbnail", "");
+
+			// This happens with the new way we are serving css files
+			if (!File.Exists(r))
+			{
+				var fileName = r;
+				var pos = fileName.LastIndexOfAny(new[] { '\\', '/' });
+				if (pos > -1) fileName = fileName.Substring(pos + 1);
+
+				var sourceDir = FileLocator.GetDirectoryDistributedWithApplication("BloomBrowserUI");
+				r = Directory.EnumerateFiles(sourceDir, fileName, SearchOption.AllDirectories).FirstOrDefault();
+			}
+
+			if (!string.IsNullOrEmpty(r) && File.Exists(r))
```

Comment:
At this point, r either existed at the start of the previous if() and presumably still does, or it is the output of EnumerateFiles. So if it's not null or empty, in what circumstances might it not exist?